lib/addons/vpc-cni/index.ts

```diff
@@ -1,15 +1,241 @@
+  */
+  disablenetworkResourceProvisioning?: boolean;
+  /**
+  * `ENABLE_POD_ENI` Environment Variable. Type: Boolean.
+  * Setting ENABLE_POD_ENI to true will allow IPAMD to add the 
+  * vpc.amazonaws.com/has-trunk-attached label to the node if the instance 
+  * has the capacity to attach an additional ENI.
+  */
+  enablePodEni?: boolean;
+  /**
+  * `ENABLE_PREFIX_DELEGATION` Environment Variable. Type: Boolean.
+  * To enable prefix delegation on nitro instances. Setting 
+  * ENABLE_PREFIX_DELEGATION to true will start allocating a prefix (/28 for IPv4 and /80 for IPv6) instead of a secondary IP in the ENIs subnet. 
+  */
+  enablePrefixDelegation?: boolean;
+  /**
+  * `WARM_ENI_TARGET` Environment Variable. Format integer.
+  * Specifies the number of free elastic network interfaces (and all of their 
+  * available IP addresses) that the ipamd daemon should attempt to keep 
+  * available for pod assignment on the node. 
+  */
+  warmEniTarget?: number;
+  /**
+  * `WARM_PREFIX_TARGET` Environment Variable. Format integer.
+  * Specifies the number of free IPv4(/28) prefixes that the ipamd daemon 
+  * should attempt to keep available for pod assignment on the node.
+  */
+  warmPrefixTarget?: number;
+  /**
+   * Secondary Subnet IDs for creating `ENIConfig`
+   */
+  subnetIds?: string[];
+  /**
+   * Corrosponding Availability Zones of the Secondary Subnet IDs 
+   * for creating `ENIConfig`
+   */
+  availabilityZones?: string[];
+}
+
+const defaultProps: CoreAddOnProps = {
+  addOnName: 'vpc-cni',
+  version: 'v1.12.1-eksbuild.2',
+  saName: 'vpc-cni',
+  namespace: 'kube-system',
+  configurationValues: {}
+};
+
+/**
+ * Implementation of VpcCni EKS add-on with Advanced Configurations.
  */
 export class VpcCniAddOn extends CoreAddOn {
+    
+      readonly vpcCniAddOnProps: VpcCniAddOnProps;
+      readonly id? : string;
 
-    constructor(version?: string) {
-        super({
-            addOnName: "vpc-cni",
-            version: version ?? "v1.12.0-eksbuild.1",
-            saName: "vpc-cni"
-        });
+      constructor(props?: VpcCniAddOnProps) {
+        super({...defaultProps, ...props});
+        this.vpcCniAddOnProps = { ...defaultProps, ...props, };
+        (this.vpcCniAddOnProps.configurationValues as any) = populateVpcCniConfigurationValues(props);
+      }
+
+      deploy(clusterInfo: ClusterInfo, props?: VpcCniAddOnProps): Promise<Construct> {
+  
+          const cluster = clusterInfo.cluster;
+          let clusterSecurityGroupId = cluster.clusterSecurityGroupId;
+          let doc: string;
+          if ((props?.subnetIds) && (props?.availabilityZones)) {
+            for (let subnetID in props?.subnetIds) {
+              doc = readYamlDocument(__dirname + '/eniConfig.yaml');
+              const manifest = doc.split("---").map(e => loadYaml(e));
+              const values: Values = {
+                  availabilityZone: this.vpcCniAddOnProps.availabilityZones![subnetID],
```

Comment:
We need to improve the design. at the moment the length of passed AZs and order must match the length of subnets and order. If customers don't follow it will fail. If they pass less AZs than subnet, the code will also fail.
Let's solve the other comment on AZs - whether they are needed and change the design to eliminate this issue.